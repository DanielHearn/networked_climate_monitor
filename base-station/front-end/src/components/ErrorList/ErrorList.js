export default {
  name: 'error-list',
  props: {
    title: {
      type: String,
      default: '',
      required: false
    },
    errors: {
      type: Array,
      default: [],
      required: true
    }
  }
}
