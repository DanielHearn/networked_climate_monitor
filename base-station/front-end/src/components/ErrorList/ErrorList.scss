@import "./../../scss/variables.scss";

.error-list {
  padding: 1em;
  display: flex;
  align-items: flex-start;
  flex-direction: column;
  color: $white;
  background: $red;
  list-style: none;
  margin: 0.5em 0;
  .title {
    font-size: $s-size;
    font-weight: $medium-weight;
  }
  p {
    margin: 0;
    font-size: $s-size;
  }
  ul {
    list-style: none;
    padding: 0;
    margin: 0;
  }
  li {

    text-align: left;
  }
}