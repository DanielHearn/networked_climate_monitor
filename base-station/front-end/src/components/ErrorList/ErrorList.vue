<template>
  <div class="error-list">
    <p v-if="title" class="title">{{ title }}</p>
    <ul>
      <li v-for="(error, index) in errors" :key="index">
        <p>{{ error }}</p>
      </li>
    </ul>
  </div>
</template>

<script src="./ErrorList.js"></script>

<style lang="scss" src="./ErrorList.scss"></style>
