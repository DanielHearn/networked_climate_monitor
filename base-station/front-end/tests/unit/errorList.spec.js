import { mount } from '@vue/test-utils'
import ErrorList from './../../src/components/ErrorList/ErrorList.vue'

describe('ErrorList.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(ErrorList, {
      propsData: {
        title: 'Errors:',
        errors: ['Error 1', 'Error 2', 'Error 3']
      }
    })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('props', () => {
    expect(wrapper.find('.title').exists()).toBe(true)
    expect(wrapper.find('.title').text()).toBe('Errors:')
    const errors = wrapper.findAll('.errors .error')
    expect(errors.length).toBe(3)
    for (let i = 0; i < errors.length; i++) {
      const element = errors.at(i)
      expect(element.text()).toBe(`Error ${i + 1}`)
    }
  })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
