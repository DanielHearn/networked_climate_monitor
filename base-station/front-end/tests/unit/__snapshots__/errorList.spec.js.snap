// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorList.vue has the expected html structure 1`] = `
<div
  class="error-list"
>
  <p
    class="title"
  >
    Errors:
  </p>
   
  <ul
    class="errors"
  >
    <li
      class="error"
    >
      <p>
        Error 1
      </p>
    </li>
    <li
      class="error"
    >
      <p>
        Error 2
      </p>
    </li>
    <li
      class="error"
    >
      <p>
        Error 3
      </p>
    </li>
  </ul>
</div>
`;
